pkg/cmd/target/remove.go

```diff
@@ -50,16 +50,40 @@ var targetRemoveCmd = &cobra.Command{
 			selectedTargetName = args[0]
 		}
 
+		ctx := context.Background()
 		client, err := apiclient.GetApiClient(nil)
 		if err != nil {
 			log.Fatal(err)
 		}
 
-		res, err := client.TargetAPI.RemoveTarget(context.Background(), selectedTargetName).Execute()
+		res, err := client.TargetAPI.RemoveTarget(ctx, selectedTargetName).Execute()
 		if err != nil {
 			log.Fatal(apiclient.HandleErrorResponse(res, err))
 		}
 
+		workspaceList, res, err := client.WorkspaceAPI.ListWorkspaces(ctx).Execute()
+		if err != nil {
+			log.Error(apiclient.HandleErrorResponse(res, err))
+		}
+
+		if len(workspaceList) > 0 {
+			views.RenderInfoMessage(fmt.Sprintln("Deleting workspaces within target..."))
+			for _, workspace := range workspaceList {
+				if *workspace.Target != selectedTargetName {
+					continue
+				}
+
+				res, err := client.WorkspaceAPI.RemoveWorkspace(ctx, *workspace.Id).Execute()
+				if err != nil {
```

Comment:
Workspace removal should be done before the target is removed.

If the target is removed from the Daytona Server, workspaces can not be removed and removal will fail with the following error:
```
ERRO[0000] Failed to delete workspace WORKSPACE_NAME: failed to remove workspace: record not found
```